src/main/java/dev/jbang/cli/Info.java

```diff
@@ -69,52 +81,50 @@ static class ScriptInfo {
 		String availableJdkPath;
 		List<String> compileOptions;
 		List<String> runtimeOptions;
-		List<ResourceFile> files;
-		List<ScriptInfo> sources;
+		List<ProjectFile> files;
+		List<ProjectFile> sources;
 		String description;
 		String gav;
 
-		public ScriptInfo(Code code, RunContext ctx) {
-			originalResource = code.getResourceRef().getOriginalResource();
+		public ScriptInfo(Project prj, ProjectBuilder pb) {
+			originalResource = prj.getResourceRef().getOriginalResource();
 
 			if (scripts.add(originalResource)) {
-				backingResource = code.getResourceRef().getFile().toString();
+				backingResource = prj.getResourceRef().getFile().toString();
 
-				init(code.asProject());
+				init(prj);
 
-				if (ctx != null) {
-					applicationJar = code.getJarFile() == null ? null : code.getJarFile().toAbsolutePath().toString();
-					mainClass = ctx.getMainClassOr(code);
-					requestedJavaVersion = code.getJavaVersion();
+				if (pb != null) {
+					applicationJar = prj.getJarFile() == null ? null : prj.getJarFile().toAbsolutePath().toString();
+					nativeImage = prj.getNativeImageFile() == null || !Files.exists(prj.getNativeImageFile()) ? null
+							: prj.getNativeImageFile().toAbsolutePath().toString();
+					mainClass = prj.getMainClass();
+					requestedJavaVersion = prj.getJavaVersion();
 					availableJdkPath = Objects.toString(JdkManager.getCurrentJdk(requestedJavaVersion), null);
 
-					String cp = ctx.resolveClassPath(code).getClassPath();
+					String cp = prj.resolveClassPath().getClassPath();
 					if (cp.isEmpty()) {
 						resolvedDependencies = Collections.emptyList();
 					} else {
 						resolvedDependencies = Arrays.asList(cp.split(CP_SEPARATOR));
 					}
 
-					if (code.asJar() != null && code.asJar().getBuildJdk() > 0) {
-						javaVersion = Integer.toString(code.asJar().getBuildJdk());
+					// TODO remove if everything okay
```

Comment:
hmm - what did that code ensure in the past? when jdk was stated inside the jar it would use that instead of default?